pkg/manifests/manifests_test.go

```diff
@@ -426,16 +432,146 @@ func TestGetDaemonSet(t *testing.T) {
+				rteNotifierVolumeName:           fmt.Sprintf("/%s", rteNotifierVolumeName),
+			},
+		},
+		{
+			name: "Verify DaemonSet generation for Kubernetes platform",
+			plat: platform.Kubernetes,
+			expectedCommandArgs: []string{
+				fmt.Sprintf("--sysfs=%s", containerHostSysDir),
+				fmt.Sprintf("--podresources-socket=unix://%s", containerPodResourcesSocket),
+				fmt.Sprintf("--kubelet-config-file=/%s/config.yaml", rteKubeletDirVolumeName),
+				fmt.Sprintf("--kubelet-state-dir=/%s", rteKubeletDirVolumeName),
+			},
+			expectedVolumes: map[string]string{
+				rteSysVolumeName:                "/sys",
+				rtePodresourcesSocketVolumeName: "/var/lib/kubelet/pod-resources/kubelet.sock",
+				rteKubeletDirVolumeName:         "/var/lib/kubelet",
+			},
+			expectedVolumeMounts: map[string]string{
+				rteSysVolumeName:                containerHostSysDir,
+				rtePodresourcesSocketVolumeName: containerPodResourcesSocket,
+				rteKubeletDirVolumeName:         fmt.Sprintf("/%s", rteKubeletDirVolumeName),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ds, err := DaemonSet(ComponentResourceTopologyExporter, tc.plat, "test")
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			// we are expecting 3 volumes
+			// 1. Host sys
+			// 2. Pod resources socket file
+			// 3. RTE notifier directory for OpenShift platform or /var/lib/kubelet for Kubernetes platform
+			if len(ds.Spec.Template.Spec.Volumes) != 3 {
+				klog.Errorf("the daemon set volumes: %+v", ds.Spec.Template.Spec.Volumes)
+				t.Fatalf("the daemon set has %d volumes when it should have %d", len(ds.Spec.Template.Spec.Volumes), 2)
+			}
+
+			for _, v := range ds.Spec.Template.Spec.Volumes {
+				path, ok := tc.expectedVolumes[v.Name]
+				if !ok {
+					t.Fatalf("the volume %q does not exist under expected volumes %v", v.Name, tc.expectedVolumes)
 				}
-			} else {
-				if err != nil || obj == nil {
-					t.Fatalf("nil obj or non-nil err=%v", err)
+
+				if v.HostPath.Path != path {
+					t.Fatalf("the volume %q path %q does not have expected value %q", v.Name, v.HostPath.Path, path)
+				}
+			}
+
+			rteContainer := ds.Spec.Template.Spec.Containers[0]
+			if len(rteContainer.VolumeMounts) != 3 {
+				klog.Errorf("the daemon set container volume mounts: %+v", rteContainer.VolumeMounts)
+				t.Fatalf("the daemon set container has %d volume mounts when it should have %d", len(rteContainer.VolumeMounts), 2)
+			}
+
+			for _, m := range rteContainer.VolumeMounts {
+				path, ok := tc.expectedVolumeMounts[m.Name]
+				if !ok {
+					t.Fatalf("the volume mount %q does not exist under expected volumes mounts %v", m.Name, tc.expectedVolumeMounts)
+				}
+
+				if m.MountPath != path {
+					t.Fatalf("the volume mount %q path %q does not have expected value %q", m.Name, m.MountPath, path)
+				}
+			}
+
+			containerCommand := strings.Join(rteContainer.Command, " ")
```

Comment:
this is a bit fragile, but I don't have better suggestions so let's keep like this for the time being